src/taco-dependency-installer/test/javaJdkInstaller.ts

```diff
@@ -0,0 +1,344 @@
+
+        tacoUtils.Telemetry.init("TACO/dependencyInstaller", "1.2.3", false);
+
+        // Register mocks. child_process and taco-utils mocks needs to be registered before 
+        // javaJdkInstaller is required for the mocking to work
+        childProcessModule = new nodeFakes.ChildProcessModule();
+        mockery.registerMock("child_process", childProcessModule);
+
+        // Reload taco-tests-utils but now with the fake processUtils loaded, so the fake telemetry will use the fake process
+        var tacoTestsUtilsWithMocks: typeof tacoTestsUtils = require("taco-tests-utils");
+
+        fakeTelemetryHelper = new tacoTestsUtilsWithMocks.TelemetryFakes.Helper();
+        var tacoUtilsWithFakes = _.extend({}, tacoUtils, { TelemetryHelper: fakeTelemetryHelper, HasFakes: true }, fakeProcessUtilsModule);
+        mockery.registerMock("taco-utils", tacoUtilsWithFakes); // javaJdkInstaller loads taco-utils
+
+        // We require the javaJdkInstaller file, which will use all the mocked dependencies
+        javaJdkInstallerModule = require("../installers/javaJdkInstaller");
+    });
+
+    after(() => {
+        // Clean up and revert everything back to normal
+        mockery.deregisterAll();
+        mockery.disable();
+    });
+
+    beforeEach(() => {
+        fakeTelemetryHelper.clear(); // So we'll only get the new events in each scenario
+        steps = { download: false, install: false, updateVariables: false, postInstall: false }; // We reset all the steps to false
+        installTo = "C:\\Program Files (x86)\\JDK"; // Default installation directory in windows
+    });
+
+    function telemetryGeneratedShouldBe(expectedTelemetry: TacoUtility.ICommandTelemetryProperties[],
+        expectedMessagePattern: RegExp, done: MochaDone): void {
+        var javaJdkInstaller: JavaJdkInstaller = new javaJdkInstallerModule(installerInfo, softwareVersion, installTo,
+            new FakeLogger(), steps);
+
+        return javaJdkInstaller.run()
+            .then<any>(() => Q.reject(new Error("Should have gotten a rejection in this test")), (error: Error) => {
+                return fakeTelemetryHelper.getAllSentEvents().then((allSentEvents: TelemetryEvent[]) => {
+                    // We check the message first, because some coding defects can make the tests end in unexpected states
+                    error.message.should.match(expectedMessagePattern);
+
+                    // Then we validate the telemetry
+                    allSentEvents.should.eql(expectedTelemetry);
+                });
+            }).done(() => done(), done);
+    }
+
+    describe("installation in mac", () => {
+        enum ChildProcessExecBehavior {
+            PackageInstallerCommandFailsScenario,
+            PackageInstallerCommandCantBeRunScenario,
+            DetachingDmgFileFailsScenario
+        }
+
+        var callTime: number;
+        var childProcessExecBehavior: ChildProcessExecBehavior;
```

Comment:
Does it make sense to reset this to null in the `beforeEach()` or in a `afterEach()`? That way, if someone new adds a test case in the future and forgets to set it to the correct value in his test, we won't get undeterministic results?